dbos/system_database.py

```diff
@@ -1010,3 +1017,48 @@ def _is_buffers_empty(self) -> bool:
             len(self._workflow_status_buffer) == 0
             and len(self._workflow_inputs_buffer) == 0
         )
+
+    def enqueue(self, workflow_id: str, queue_name: str) -> None:
+        with self.engine.begin() as c:
+            c.execute(
+                pg.insert(SystemSchema.job_queue)
+                .values(
+                    workflow_uuid=workflow_id,
+                    queue_name=queue_name,
+                )
+                .on_conflict_do_nothing()
+            )
+
+    def dequeue(self, queue_name: str, concurrency: Optional[int]) -> List[str]:
+        with self.engine.begin() as c:
+            query = sa.select(SystemSchema.job_queue.c.workflow_uuid).where(
+                SystemSchema.job_queue.c.queue_name == queue_name
+            )
+            if concurrency is not None:
+                query = query.order_by(
+                    SystemSchema.job_queue.c.created_at_epoch_ms.asc()
```

Comment:
If this policy is First Come First Server, all queries must be sorted by insertion time.

nit: what you name `concurrency` here seems to be more accurately named `batch size` ?

Reply:
It's FIFO and sorted. It's called concurrency because its a bound on concurrency.

Reply:
I don't understand: if concurrency is None, then the query is

```
        query = sa.select(SystemSchema.job_queue.c.workflow_uuid).where(
                SystemSchema.job_queue.c.queue_name == queue_name
            )
```

which doesn't sort the results. So it's not FIFO?

Reply:
And how is it a bound on concurrency? This is about getting a hold on items on the queue, this is not about the processing of these items.